### PR TITLE
Merge Network's release branch into master

### PR DESCRIPTION
**Pull request by anton-evseev**
## Description
This is `network-april` + commit to resolve merge conflicts

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by number213** [comment]
@cormacpayne @MiYanni could you please take a look? This is `network-april` + commit to resolve merge conflicts
cc: @MikhailTryakhov

File: tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv [hunk 1]
```diff
@@ -447,3 +447,57 @@
...
 "Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.SetAzureVirtualNetworkGatewayCommand","Set-AzVirtualNetworkGateway","0","1050","The parameter set '__AllParameterSets' for cmdlet 'Set-AzVirtualNetworkGateway' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Set-AzVirtualNetworkGateway'."
+"Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.AddAzureExpressRouteCircuitAuthorizationCommand","Add-AzExpressRouteCircuitAuthorization","0","3000","The type of property 'RouteFilter' of type 'Microsoft.Azure.Commands.Network.Models.PSPeering' has changed from 'Microsoft.Azure.Commands.Network.Models.PSRouteFilter' to 'Microsoft.Azure.Commands.Network.Models.PSResourceId'.","Change the type of property 'RouteFilter' back to 'Microsoft.Azure.Commands.Network.Models.PSRouteFilter'."
```
Comment (markcowl) [inline tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv]: What are these breaking changes in PSRouteFilter?

I understand the removal of the InterfaceEndpoints feature, but is this related.  Breaking changes are not allowed.

**Comment by markcowl** [comment]
@number213 @MikhailTryakhov @chandrasekarsrinivasan Please see my comment on the addiitional breaking change beyond InterfaceEndpooints.  Is this related, or is it a separate break?

As you know breaking changes are not allowed.  In the future, we will not accept breaking changes of either kind - APIs for whitelisted features should be in a preview module, and not in the stable Network module.

File: tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv (hunk 1 above)
Comment (number213) [reply tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv]: @markcowl type of RouteFilter property has been adjusted to reflect bugfix in Swagger. Please see discussion of this change with Cormac here: https://github.com/Azure/azure-powershell/pull/9250#discussion_r286584813

**Comment by number213** [comment]
@markcowl addressed your comment, please take a look

File: tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv (hunk 1 above)
Comment (markcowl) [reply tools/StaticAnalysis/Exceptions/Az.Network/BreakingChangeIssues.csv]: got it, if the actually filled-in properties are the same, this is OK.

**Pull request merged**